docs/tutorials/on-device-training/ios-app.md

```diff
@@ -0,0 +1,987 @@
+        }
+        ortSession = try ORTSession(env: ortEnv, modelPath: modelPath, sessionOptions: nil)
+    }
+}
+```
+
+Next, we will write the `evaluate` method that will take in the audio data, convert it to `ORTValue`, and perform inference with the model. Then, take the output of the model and extract probabilities.
+
+
+```swift
+    private func isUser(logits: [Float]) -> Float {
+        // apply softMax
+        let maxInput = logits.max() ?? 0.0
+        let expValues = logits.map { exp($0 - maxInput) } // Calculate e^(x - maxInput) for each element
+        let expSum = expValues.reduce(0, +) // Sum of all e^(x - maxInput) values
+        
+        return expValues.map { $0 / expSum }[1] // Calculate the softmax probabilities
+    }
+    
+    func evaluate(inputData: Data) -> Result<(Bool, Float), Error> {
+        
+        return Result<(Bool, Float), Error> { () -> (Bool, Float) in
+            
+            // convert input data to ORTValue
+            let inputShape: [NSNumber] = [1, inputData.count / MemoryLayout<Float>.stride as NSNumber]
+            
+            let input = try ORTValue(
+                tensorData: NSMutableData(data: inputData),
+                elementType: ORTTensorElementDataType.float,
+                shape: inputShape)
+            
+            let outputs = try ortSession.run(
+                withInputs: ["input": input],
+                outputNames: ["output"],
+                runOptions: nil)
+            
+            guard let output = outputs["output"] else {
+                throw VoiceIdentifierError.Error("Failed to get model output from inference.")
+            }
+            
+            let outputData = try output.tensorData() as Data
+            let probUser = outputData.withUnsafeBytes { (buffer: UnsafeRawBufferPointer) -> Float in
+                let floatBuffer = buffer.bindMemory(to: Float.self)
+                let logits = Array(UnsafeBufferPointer(start: floatBuffer.baseAddress, count: outputData.count/MemoryLayout<Float>.stride))
+                return isUser(logits: logits)
+            }
+            
+            return (probUser >= kThresholdProbability, probUser)
+        }
+    }
+```
+
+The complete implementation of the `VoiceIdentifier` class can be found [here](https://github.com/microsoft/onnxruntime-training-examples/blob/master/on_device_training/mobile/ios/MyVoice/VoiceIdentifier.swift)
+
+### Recording Audio
+
+We will use the `AudioRecorder` class to record audio through the microphone. It will record 10 seconds of audio and output the audio data as a `Data` object, which can be used for training and inference purposes. we will use `AVFoundation` framework to access microphone and record the audio. There will be one public method `record(callback: @escaping RecordingDoneCallback)` that will record the audio and call the callback function with the audio data as `Data` object, when the recording is done.
```

Comment:
```suggestion
We will use the `AudioRecorder` class to record audio through the microphone. It will record 10 seconds of audio and output the audio data as a `Data` object, which can be used for training and inference purposes. We will use the `AVFoundation` framework to access the microphone and record the audio. There will be one public method `record(callback: @escaping RecordingDoneCallback)` that will record the audio and call the callback function with the audio data when the recording is done.
```